gqlalchemy/memgraph.py

```diff
@@ -585,3 +585,36 @@ def get_lambda(expression: str) -> str:
             return ""
         else:
             return f"(e, v | {expression})"
+
+
+class WeightedShortestPath(IntegratedAlgorithm):
+    """Build a Djikstra shortest path call for a Cypher query"""
+
+    def __init__(
+        self,
+        upper_bound: Union[int, str] = None,
+        condition: str = None,
+        total_weight_name: str = "total_weight",
```

Comment:
I would use `total_weight` instead of `total_weight_name`. Or if you don't want to use total_weight because you don't want the user to think it's the value, maybe then use `total_weight_variable`, or `total_weight_var`.